packages/did/src/Did.utils.ts

```diff
@@ -44,52 +45,78 @@ const KILT_DID_PREFIX = 'did:kilt:'
 // - did:kilt:<kilt_address>
 // - did:kilt:<kilt_address>#<fragment>
 const FULL_KILT_DID_REGEX =
-  /^did:kilt:(?<identifier>4[1-9a-km-zA-HJ-NP-Z]{47})(?<fragment>#[^#\n]+)?$/
+  /^did:kilt:(?<address>4[1-9a-km-zA-HJ-NP-Z]{47})(?<fragment>#[^#\n]+)?$/
 
 // Matches the following light DIDs
 // - did:kilt:light:00<kilt_address>
 // - did:kilt:light:01<kilt_address>:<encoded_details>
 // - did:kilt:light:10<kilt_address>#<fragment>
 // - did:kilt:light:99<kilt_address>:<encoded_details>#<fragment>
 const LIGHT_KILT_DID_REGEX =
-  /^did:kilt:light:(?<auth_key_type>[0-9]{2})(?<identifier>4[1-9a-km-zA-HJ-NP-Z]{47,48})(?<encoded_details>:.+?)?(?<fragment>#[^#\n]+)?$/
+  /^did:kilt:light:(?<authKeyType>[0-9]{2})(?<address>4[1-9a-km-zA-HJ-NP-Z]{47,48})(:(?<encodedDetails>.+?))?(?<fragment>#[^#\n]+)?$/
 
 export const defaultKeySelectionCallback = <T>(keys: T[]): Promise<T | null> =>
   Promise.resolve(keys[0] || null)
 
+/**
+ * Extracts the KILT address from the DID identifier, even if it is the light DID.
+ *
+ * @param identifier The identifier of light or full DID.
+ * @returns The KILT address.
+ */
+export function getAddressFromIdentifier(
+  identifier: DidIdentifier
+): KiltAddress {
+  const isAddress = checkAddress(identifier, ss58Format)[0]
+  const address = isAddress ? identifier : identifier.substring(2)
+  return address as KiltAddress
+}
+
 /**
  * Compiles a KILT DID uri for a full or light DID from a unique identifier and associated data.
  *
  * @param identifier A ss58 encoded address valid on the KILT network.
- * @param didType 'full' to produce a FullDid's uri, 'light' for a LightDid.
+ * @param type 'full' to produce a FullDid's uri, 'light' for a LightDid.
  * @param version KILT DID specification version number.
  * @param encodedDetails When compiling a LightDid uri, encoded DidDetails can be appended to the end of the uri.
  * @returns A DID uri as a string.
  */
 export function getKiltDidFromIdentifier(
   identifier: DidIdentifier,
```

Comment:
Is there a distinction between an identifier and an address here?
In the regex above it was already renamed and I think it is better to be consistent with the terminology, if possible.

Reply:
The address is always `4...`. The identifier in the light DID could be `004...`, it’everything after `did:kilt:` prefix. The function was created before I put in place the proper distinction between the address and identifier.

I will review its usages to see what it should be.

Reply:
So I took a long look at the usages of `getKiltDidFromIdentifier()`. Many of those we for the migration flow: I have stored a light DID and want to know its new URI. I have added a dedicated function for this purpose: `getFullDidUri()`. Besides, some code gets an address from the blockchain and needs to transform that to a full DID, and the same function helps with that. There are some tests that hard-code the address but not the DID, why not both? Finally, there was some raw string processing that could be simplified. After all these updates, I was able to remove the `getKiltDidFromIdentifier()` completely.